### PR TITLE
docs: Add initial qemu documentation

### LINKED ISSUE #2851: Add qemu documentation
The `qemu` backend is one of the most feature full backend but has approximately zero documentation

### PR DESCRIPTION
**Pull request by rn**
resolves #2851

bear-wings

### CODE REVIEW

File: docs/platform-qemu.md [hunk 1]
```diff
@@ -1,15 +1,37 @@
 # LinuxKit with qemu/kvm
 
-TBD
+The `qemu` backend is the most versatile `run` backend for
+`linuxkit`. It can boot both `x86_64` and `arm64` images, runs on
+macOS and Linux (and possibly Windows), and can boot most types of
+output formats.
+
```
Comment (ijc) [inline docs/platform-qemu.md]: Do we generally mention when a backend is the default on a platform (as qemu is for Linux)? Should we?

Comment (rn) [reply docs/platform-qemu.md]: we don't for hyperkit and hyper-v which are the only other defaults

File: docs/platform-qemu.md [hunk 2]
```diff
@@ -21,13 +43,23 @@ supports a number of different disk formats.
...
+on the platform you are running. The default is the userspace
+networking which provides the VM with a internal DHCP server and
+network connectivity, but does not provide access to the VMs network
+from the outside.
+
+With user mode networking you can publish selected VM ports on the
+host, using the `-publish` option. It uses the same syntax as the
+`qemu` binary.
+
+On Linux, you can attach the VM either to an existing bridge or tap interface.
```
Comment (ijc) [inline docs/platform-qemu.md]: You didn't fancy explaining how ;-) You could perhaps steal a bunch of useful words from the `linuxkit/kubernetes/README.md`?

File: docs/platform-qemu.md [hunk 3]
```diff
@@ -21,13 +43,23 @@ supports a number of different disk formats.
...
 ## Integration services and Metadata
 
-TBD: Integration services
+The `qemu` backend also allows passing custom userdata into the
+[metadata package](./metadata.md) using either the `-data` or `-data-file` command-line
+option. This attaches a CD device with the data on.
 
-The qemu backend currently does not support passing metadata/userdata
-to the VM.
+TBD: Integration services
```
Comment (ijc) [inline docs/platform-qemu.md]: Does "Integration services" refer to the use of `pkg/qemu-ga`?

File: docs/platform-qemu.md (hunk 1 above)
Comment (ijc) [reply docs/platform-qemu.md]: Also, do we/should we highlight here or elsewhere that this stuff is intended for dev test and not production usage?

File: docs/platform-qemu.md (hunk 3 above)
Comment (rn) [reply docs/platform-qemu.md]: yes

File: docs/platform-qemu.md (hunk 2 above)
Comment (rn) [reply docs/platform-qemu.md]: Indeed. But will take a look at the kube docs. Thanks for the reminder

**Comment by rn** [comment]
Pushed an update with networking examples and mention of `qemu-ga` package

File: docs/platform-qemu.md [hunk 4]
```diff
@@ -21,13 +42,32 @@ supports a number of different disk formats.
...
+
+With user mode networking you can publish selected VM ports on the
+host, using the `-publish` option. It uses the same syntax as the
+`qemu` binary. For example `linuxkit run qemu -publish 8080:80
+linuxkit` exposes port `80` from the VM as port `8080` on the host.
+
+On Linux, you can attach the VM either to an existing bridge or tap
+interface. These require root privileges and you may want to use the
+[`qemu-bridge-helper`](http://wiki.qemu.org/Features/HelperNetworking). To
+attach to an existing bridge `br0` (e.g., one create with
```
Comment (ijc) [inline docs/platform-qemu.md]: `created`

**Pull request merged**